### PR TITLE
feat: core sqlite workers (part 1)

### PR DESCRIPTION
**Pull request by fontanierh**
[Design Doc](https://www.notion.so/dust-tt/Engineering-e0f834b5be5a43569baaf76e9c41adf2?p=e9233cd287334e439751366ff3ec8093&pm=s)

### core
- new database model for `SqliteWorker`s
- endpoints to register & delete a `SqliteWorker`

### core-sqlite-worker
- new cargo bin
- heartbeat / delete logic to register with `core` (heartbeat loop + graceful exit code)
- rough skeleton of the overall flow: receive request for a database, spin up a thread if the db is not already running, send a message to the DB

## Next steps:

- new database for `core-sqlite-worker`
- move ownership of `databases_tables` and `databases_rows` to `core-sqlite-worker`
- move querying logic to `core-sqlite-worker`

### CODE REVIEW

File: core/src/sqlite_database.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+use anyhow::{anyhow, Result};
```
Comment (spolu) [inline core/src/sqlite_database.rs:1]: why put it in `src/` directly?

File: core/src/sqlite_workers.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+use anyhow::{anyhow, Result};
+
```
Comment (spolu) [inline core/src/sqlite_workers.rs:2]: why put this file in `src/` directly?

File: core/src/http_utils.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+use axum::Json;
```
Comment (spolu) [inline core/src/http_utils.rs:1]: we can probably have that in `src/util.rs` ?

**Review by spolu — COMMENTED** [review]
LGTM

I think we want to move code around a bit because there's no reason to have both sqlite_database.rs and sqlite_workers.rs in `src/` directly. Let's move them in a sqlite_worker directly at least?

How will you get rid of database record for processes that crashed? Shouldn't you check liveness + clean-up in `core` ?

File: core/src/sqlite_database.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/sqlite_database.rs:1]: Will move into a `sqlite_worker` dir 👍

File: core/src/sqlite_workers.rs (hunk 2 above)
Comment (fontanierh) [reply core/src/sqlite_workers.rs:2]: Will move into a `sqlite_worker` dir 👍

File: core/src/http_utils.rs (hunk 3 above)
Comment (fontanierh) [reply core/src/http_utils.rs:1]: Fair 👍

**Comment by fontanierh** [comment]
> How will you get rid of database record for processes that crashed? Shouldn't you check liveness + clean-up in core ?

Not implemented yet, but indeed there should be some cleanup code that runs periodically to cleanup the dead rows.

**Comment by fontanierh** [comment]
Addressed the feedback @spolu, except the `core` `SqliteWorkers` crashed workers cleanup (will address this one in a different PR)

**Review by spolu — APPROVED** [review]
Let's make sure to address it quickly as this is the kind of things that we leave behind easily :)

**Comment by fontanierh** [comment]
> Let's make sure to address it quickly as this is the kind of things that we leave behind easily :)

👍

**Pull request merged**